cores/arduino/Serial.cpp

```diff
@@ -61,9 +61,13 @@ void UART::WrapperCallback(uart_callback_args_t *p_args) {
       case UART_EVENT_TX_COMPLETE:
       case UART_EVENT_TX_DATA_EMPTY:
       {
-        //uint8_t to_enqueue = uart_ptr->txBuffer.available() < uart_ptr->uart_ctrl.fifo_depth ? uart_ptr->txBuffer.available() : uart_ptr->uart_ctrl.fifo_depth;
-        //while (to_enqueue) {
-        uart_ptr->tx_done = true;
+          if(uart_ptr->txBuffer.available()){
+              static char txc;
+              txc = uart_ptr->txBuffer.read_char();
```

Comment:
Is it OK to use a static variable here instead of uart_ptr->txc? Would this work correctly if there are multiple uarts working simultaneously?

Reply:
That's a good question.  I made it static because it was being passed as a pointer on the next line but I didn't really think about it being there in multiple instances.  

I *think* that it will be OK because R_SCI_UART_Write is going to use that data one line later and then we're done with it.  I think it would be OK to make it not static.  

I wish I could grab a whole chunk of the tx_buffer there so the code in r_sci_uart.c could make effective use of the FIFO.  The problem is that I would need to know where the data in the tx_buffer wraps around or I could run off the end. 

I have a separate branch that uses DMA and it doesn't have this problem because I set the DMA extended repeat area to match the tx_buffer and the DMA controller will automatically wrap back around.  But that code seems to be broken in other places.

Reply:
Actually, looking at it now, I don't see why I don't just use `uart_ptr->txc` instead.  Maybe I thought because it was an interrupt handler that I wouldn't have access but I do.  

I'll update the branch tonight and retest a little.

Reply:
I updated the code to use the member variable instead.  It tests ok.  I did a lot of printing without any failures.  Unfortunately I seem to have messed up my branch.

Reply:
Updated once more to load characters 16 at a time in the interrupt callback.  This takes advantage of the FIFO to reduce the number of interrupts.